src/api/flaskr/service/learn/ask_provider_adapters.py

```
@@ -0,0 +1,310 @@
+        if not isinstance(config, dict):
+            config = {}
+
+        bot_id = str(config.get("bot_id") or "").strip()
+        api_path = str(config.get("api_path") or "/v3/chat").strip() or "/v3/chat"
+        if api_path.startswith("http"):
+            url = api_path
+        else:
+            url = base_url.rstrip("/") + "/" + api_path.lstrip("/")
+
+        if api_path == "/v3/chat" and not bot_id:
+            raise AskProviderConfigError("coze bot_id is required")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)